Try to fix JOGL on Linux (use newt)

deathbeam opened the pull request:
Signed-off-by: Tomas Slusny <slusnucky@gmail.com>

Fixes #6480

matt2134711 commented:
Still have broken input here :(

Also the client would not shut down cleanly, I had to `kill -9` it.

***

#### Log

```
2018-11-16 21:55:30 [main] INFO  n.r.client.account.SessionManager - No session file exists
2018-11-16 21:55:31 [pool-2-thread-1] WARN  n.runelite.http.api.item.ItemClient - Error looking up prices: Not Found
2018-11-16 21:55:31 [main] INFO  n.r.c.plugins.kourendlibrary.Library - Library is now reset
2018-11-16 21:55:31 [main] WARN  n.r.client.ClientSessionManager - error opening session
java.io.IOException: java.lang.IllegalArgumentException: Invalid UUID string: <html><head><title>Apache
	at net.runelite.http.api.session.SessionClient.open(SessionClient.java:59)
	at net.runelite.client.ClientSessionManager.start(ClientSessionManager.java:58)
	at net.runelite.client.RuneLite.start(RuneLite.java:258)
	at net.runelite.client.RuneLite.main(RuneLite.java:221)
Caused by: java.lang.IllegalArgumentException: Invalid UUID string: <html><head><title>Apache
	at java.util.UUID.fromString(UUID.java:194)
	at com.google.gson.internal.bind.TypeAdapters$24.read(TypeAdapters.java:550)
	at com.google.gson.internal.bind.TypeAdapters$24.read(TypeAdapters.java:543)
	at com.google.gson.Gson.fromJson(Gson.java:927)
	at com.google.gson.Gson.fromJson(Gson.java:865)
	at net.runelite.http.api.session.SessionClient.open(SessionClient.java:55)
	... 3 common frames omitted
2018-11-16 21:55:32 [AWT-EventQueue-1] INFO  net.runelite.client.ui.ClientUI - Showing frame net.runelite.client.ui.ContainableFrame[frame0,556,273,809x534,layout=java.awt.BorderLayout,title=RuneLite,resizable,maximized,defaultCloseOperation=DO_NOTHING_ON_CLOSE,rootPane=javax.swing.JRootPane[,0,0,809x534,layout=org.pushingpixels.substance.internal.ui.SubstanceRootPaneUI$SubstanceRootLayout,alignmentX=0.0,alignmentY=0.0,border=org.pushingpixels.substance.internal.utils.border.SubstancePaneBorder@1f79e28d,flags=16777673,maximumSize=,minimumSize=,preferredSize=],rootPaneCheckingEnabled=true]
2018-11-16 21:55:32 [main] INFO  n.r.client.discord.DiscordService - Initializing Discord RPC service.
2018-11-16 21:55:33 [main] INFO  net.runelite.client.RuneLite - Client initialization took 3286ms. Uptime: 3509ms
2018-11-16 21:55:33 [Thread-4] WARN  net.runelite.client.RuneLiteModule - uncaught exception in event subscriber
java.lang.NullPointerException: null
	at net.runelite.client.plugins.defaultworld.DefaultWorldPlugin.changeWorld(DefaultWorldPlugin.java:114)
	at net.runelite.client.plugins.defaultworld.DefaultWorldPlugin.applyWorld(DefaultWorldPlugin.java:149)
	at net.runelite.client.plugins.defaultworld.DefaultWorldPlugin.onGameStateChanged(DefaultWorldPlugin.java:91)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.google.common.eventbus.Subscriber.invokeSubscriberMethod(Subscriber.java:91)
	at com.google.common.eventbus.Subscriber$SynchronizedSubscriber.invokeSubscriberMethod(Subscriber.java:150)
	at com.google.common.eventbus.Subscriber$1.run(Subscriber.java:76)
	at com.google.common.util.concurrent.MoreExecutors$DirectExecutor.execute(MoreExecutors.java:399)
	at com.google.common.eventbus.Subscriber.dispatchEvent(Subscriber.java:71)
	at com.google.common.eventbus.Dispatcher$PerThreadQueuedDispatcher.dispatch(Dispatcher.java:116)
	at com.google.common.eventbus.EventBus.post(EventBus.java:217)
	at net.runelite.client.callback.Hooks.post(Hooks.java:128)
	at client.gameStateChanged(client.java:914)
	at s.eb(l.java:1376)
	at u.ek(u.java:2068)
	at client.ad(client.java:886)
	at ah.ac(ah.java:338)
	at ah.run(ah.java:317)
	at java.lang.Thread.run(Thread.java:748)
libEGL warning: DRI2: failed to authenticate
X11Util.Display: Shutdown (JVM shutdown: true, open (no close attempt): 1/1, reusable (open, marked uncloseable): 0, pending (open in creation order): 1)
X11Util: Open X11 Display Connections: 1
X11Util: Open[0]: NamedX11Display[:1, 0x7f9570810b50, refCount 1, unCloseable false]
2018-11-16 21:56:31 [pool-2-thread-1] WARN  n.r.client.ClientSessionManager - null
java.io.IOException: java.lang.IllegalArgumentException: Invalid UUID string: <html><head><title>Apache
	at net.runelite.http.api.session.SessionClient.open(SessionClient.java:59)
	at net.runelite.client.ClientSessionManager.ping(ClientSessionManager.java:93)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.IllegalArgumentException: Invalid UUID string: <html><head><title>Apache
	at java.util.UUID.fromString(UUID.java:194)
	at com.google.gson.internal.bind.TypeAdapters$24.read(TypeAdapters.java:550)
	at com.google.gson.internal.bind.TypeAdapters$24.read(TypeAdapters.java:543)
	at com.google.gson.Gson.fromJson(Gson.java:927)
	at com.google.gson.Gson.fromJson(Gson.java:865)
	at net.runelite.http.api.session.SessionClient.open(SessionClient.java:55)
	... 8 common frames omitted
```

#### System Info

```
$ java -version
openjdk version "10.0.2" 2018-07-17
OpenJDK Runtime Environment (build 10.0.2+13-Debian-2)
OpenJDK 64-Bit Server VM (build 10.0.2+13-Debian-2, mixed mode)

$ nvidia-settings --version
nvidia-settings:  version 390.67

$ git describe 
runelite-parent-1.4.26-9-g4be87b10d
```

deathbeam commented:
Updated the PR to somewhat work with NEWT window thanks to @Whitehooder. + added input proxy, so input somewhat works too (just mouse, keyboard is still broken), WIP

Adam- commented:
Superceeded by #6530
